### PR TITLE
Remove all deprecated matchers/assertions

### LINKED ISSUE #651: Remove old shouldBe methods
For 3.3 are we removing the deprecated shouldBe methods that have been deprecated since 3.0 ?
@Kerooker

### PR DESCRIPTION
**Pull request by LeoColman**
This commit removes all `shouldBe` kind of matchers/assertions that were deprecated. Some were already moved to another package, some are not the correct usage anymore.

They're deprecated quite some time, and are now being removed.

Closes #651

### CODE REVIEW

**Comment by LeoColman** [comment]
I don't think there is any other matcher to remove

**Comment by LeoColman** [comment]
But we can discuss if we want to remove `isInstancePerTest` and the deprecated TestListener functions

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/StringMatchers.kt [hunk 1]
```diff
@@ -23,8 +23,6 @@ fun startWith(prefix: String) = neverNullMatcher<String> { value ->
 }
 
 fun haveSubstring(substr: String) = include(substr)
-@Deprecated("use should include(substring)", ReplaceWith("include(substr)", "io.kotlintest.matchers.string.include"))
```
Comment (sksamuel) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/StringMatchers.kt:26]: How long has this one been deprecated for ?

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt [hunk 2]
```diff
@@ -62,11 +51,6 @@ fun <T> List<T>.shouldHaveElementAt(index: Int, element: T) = this should haveEl
 
 fun <T> List<T>.shouldNotHaveElementAt(index: Int, element: T) = this shouldNot haveElementAt(index, element)
 
-@Deprecated("Use shouldHaveElementAt", ReplaceWith("this.shouldHaveElementAt(index, element)", "io.kotlintest.should"))
-fun <T, L : List<T>> L.shouldContainElementAt(index: Int, element: T) = this should haveElementAt(index, element)
-
-@Deprecated("Use shouldNotHaveElementAt", ReplaceWith("this.shouldNotHaveElementAt(index, element)", "io.kotlintest.should"))
-fun <T, L : List<T>> L.shouldNotContainElementAt(index: Int, element: T) = this shouldNot haveElementAt(index, element)
```
Comment (sksamuel) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt:69]: Are these relatively recent deprecations ?

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt [hunk 3]
```diff
@@ -62,11 +51,6 @@ fun <T> List<T>.shouldHaveElementAt(index: Int, element: T) = this should haveEl
 
 fun <T> List<T>.shouldNotHaveElementAt(index: Int, element: T) = this shouldNot haveElementAt(index, element)
 
-@Deprecated("Use shouldHaveElementAt", ReplaceWith("this.shouldHaveElementAt(index, element)", "io.kotlintest.should"))
-fun <T, L : List<T>> L.shouldContainElementAt(index: Int, element: T) = this should haveElementAt(index, element)
```
Comment (sksamuel) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt:66]: Are these relatively recent deprecations ?

**Review by sksamuel — CHANGES_REQUESTED** [review]

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt (hunk 2 above)
Comment (LeoColman) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt:69]: 09/05/2018

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt (hunk 3 above)
Comment (LeoColman) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt:66]: 05/2018

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/StringMatchers.kt (hunk 1 above)
Comment (LeoColman) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/StringMatchers.kt:26]: 04/2018

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt (hunk 3 above)
Comment (sksamuel) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/collections/matchers.kt:66]: So all these deprecations went into 3.1.0 so I think we can remove
I just don't want to remove anything we only deprecated in 3.2.0 as we only released that a month ago.
And we released 3.1.0 on 11th May - so anything deprecated after that might not have been seen by a user until Jan this year.

**Comment by sksamuel** [comment]
> But we can discuss if we want to remove `isInstancePerTest` and the deprecated TestListener functions

I think no, as they only got deprecated a month ago. 3.4 for those.

**Comment by LeoColman** [comment]
No prob. Let's merge this?

**Pull request merged**